spec/workflow/states/task_spec.rb

```diff
@@ -264,21 +264,21 @@
             expect_run_async(input, :error => "States.Timeout")
             workflow.step_nonblock
 
-            expect(ctx.next_state).to          eq("State")
+            expect(ctx.state["Name"]).to       eq("State")
```

Comment:
I would have liked not having to change any specs due to the first change moving where we step, but in this case I think the test was specific to where step_next was called rather than the actual runtime state.  The key here is that after running "step" the current state is the retried one and next_state isn't set yet.

Reply:
I agree with this change.
I am feeling the change to `step_next` is getting closer to our original run and makes running a single state the most complete.

I mentioned out of band `s/state["Name"]/state_name`

Thinking we can drop `next_state` from our tests and just test `state_name`. (another PR)